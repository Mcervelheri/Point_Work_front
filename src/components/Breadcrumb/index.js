import React from 'react';

import { Breadcrumb as AntBreadcrumb } from 'antd';
import { Link } from 'react-router-dom';

import styles from './styles.module.scss';

const Breadcrumb = ({ routes }) => {

    const renderBreadcrumbItem = route => {
        const last = routes.indexOf(route) === routes.length - 1;
        let routeName;

        if (last) {
            routeName = route.name;
        } else {
            routeName = (
                <Link to={route.path}>{route.name}</Link>
            );
        }

        return (
            <AntBreadcrumb.Item key={route.name}>{routeName}</AntBreadcrumb.Item>
        );
    };

    return (
        <AntBreadcrumb className={styles.breadcrumbContainer}>
            {routes.map(renderBreadcrumbItem)}
        </AntBreadcrumb>
    );
};

export default Breadcrumb;
