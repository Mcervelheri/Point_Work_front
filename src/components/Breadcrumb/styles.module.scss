@import "../../theme/colors.scss";
@import "../../theme/variables.scss";

.breadcrumbContainer {
    padding: 16px 0 0 24px;
}
