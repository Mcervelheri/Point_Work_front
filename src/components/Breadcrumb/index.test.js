import React from 'react';

import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Breadcrumb from './index';

const history = createMemoryHistory({
    initialEntries: [''],
});

const MOCK_PATHS = [
    {
        path: '/app',
        name: 'Home',
    },
    {
        path: '/app/contas',
        name: 'Contas',
    },
    {
        name: 'Editar SÃ³cios',
    },
    {
        path: '/app',
        name: 'Home',
    },
];

test('O Componente <Breadcrumb /> deve ser renderizado na tela corretamente', () => {
    const Element = renderer.create(
        <Router history={history}>
            <Breadcrumb
                routes={MOCK_PATHS}
            />
        </Router>,

    ).toJSON();
    expect(Element).toMatchSnapshot();
});
