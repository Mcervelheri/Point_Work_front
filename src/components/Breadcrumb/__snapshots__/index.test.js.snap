// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`O Componente <Breadcrumb /> deve ser renderizado na tela corretamente 1`] = `
<nav
  className="ant-breadcrumb breadcrumbContainer"
>
  <ol>
    <li>
      <span
        className="ant-breadcrumb-link"
      >
        <a
          href="/app"
          onClick={[Function]}
        >
          Home
        </a>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        /
      </span>
    </li>
    <li>
      <span
        className="ant-breadcrumb-link"
      >
        <a
          href="/app/contas"
          onClick={[Function]}
        >
          Contas
        </a>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        /
      </span>
    </li>
    <li>
      <span
        className="ant-breadcrumb-link"
      >
        <a
          href=""
          onClick={[Function]}
        >
          Editar SÃ³cios
        </a>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        /
      </span>
    </li>
    <li>
      <span
        className="ant-breadcrumb-link"
      >
        Home
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        /
      </span>
    </li>
  </ol>
</nav>
`;
